### PR TITLE
Add design doc for production cloud accounts + projects

### PR DESCRIPTION
**Pull request by gandelman-a**
This is a quick design doc that describes our production cloud accounts
and projects.

Related-issue: BonnyCI/projman#130
Signed-off-by: Adam Gandelman <adamg@ubuntu.com>

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/bonnyci.org/34/1487634780.3/bonnyci-run-check : FAILURE in 2m 12s

File: lore/developers/contributing/designs/production-cloud-accounts.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+We will require the following projects be created:
+
+* `bonnyci` - This project will contain cloud resources for running the production systems and provide API access to humans running automation to do initial provisioning of networks, instances and accounts.
+* `nodepool` - This is a separate project that will be configured as a Nodepool provider.  All Nodepool slaves will be booted on here.
+
+All BonnyCI developers will have a project and matching account on the cloud (see [Personal Cloud Accounts](#personal-cloud-accounts)).  Each account will also be granted a role of `_member_` on the `bonnyci` project so they may get visibility into our running systems.  This will happen en masse via a new `bonnyci` keystone group that contains all current BonnyCI developers. A subset of these users will also be assigned to a `bonnyci-admin` group and granted `cloud_admin` role on the `bonnyci` project so they may manually run automation to provision resources and manage accounts and projects.
```
Comment (omgjlk) [inline lore/developers/contributing/designs/production-cloud-accounts.md:25]: Ah, this conflicts with the PR I just put up, but I think I like your design a bit better, whereas I was giving everybody group membership in an admin group. 

How do we determine who is an admin vs who is not?

**Review by omgjlk — COMMENTED** [review]
I like the design, but I'll refrain from approving until others have a chance to read. Also I have a question.

File: lore/developers/contributing/designs/production-cloud-accounts.md (hunk 1 above)
Comment (gandelman-a) [reply lore/developers/contributing/designs/production-cloud-accounts.md:25]: I was thinking we can just grant it to people as needed.  Whenever someone wants to pick up a task that requires it, they can add a patch to request access. Over time we'll either end up with a subset of people who actually need it, or everyone in it... in which case we can just migrate to giving it to everybody

**Comment by omgjlk** [comment]
This has conflicts now :(

**Review by eggshell — COMMENTED** [review]
Looks good to me. I based BonnyCI/hoist#168 on this spec.

**Review by ConsoleCatzirl — COMMENTED** [review]
:+1: This seems fine to me, but not approving so that others can also review.

**Comment by rattboi** [comment]
yeah, lgtm as well.

**Review by omgjlk — APPROVED** [review]

**Pull request merged**